### PR TITLE
Multi: v2 staking design

### LINKED ISSUE #976: Crash on Navigation to staking
On the current master master branch Navigating to Staking crashes godcr. See log below.
`2022-06-09 14:16:16.584 [INF] DLWL: Loaded 2 wallets
2022-06-09 14:16:18.017 [INF] DLWL: Opened wallet
2022-06-09 14:16:19.199 [INF] DLWL: Opened wallet
panic: runtime error: integer divide by zero

goroutine 40 [running]:
github.com/planetdecred/godcr/ui/decredmaterial.(*MultiLayerProgressBar).progressBarLayout.func2(0xbf)
	/home/amos/go/src/github.com/planetdecred/godcr/ui/decredmaterial/progressbar.go:155 +0x6d
github.com/planetdecred/godcr/ui/decredmaterial.(*MultiLayerProgressBar).progressBarLayout.func3({{{0x0, 0x0}, {0x212, 0xf41e5}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, 0xe65ee2f75, ...}, ...})
	/home/amos/go/src/github.com/planetdecred/godcr/ui/decredmaterial/progressbar.go:162 +0x43
gioui.org/layout.Flex.Layout({0x5, 0x0, 0x0, 0x0}, {{{0x212, 0x0}, {0x212, 0xf41e5}}, {0x3f800000, 0x3f800000}, ...}, ...)
	/home/amos/go/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/decredmaterial.(*MultiLayerProgressBar).progressBarLayout(0xc000093aa0, {{{0x212, 0x0}, {0x212, 0xf41e5}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, ...}, ...})
	/home/amos/go/src/github.com/planetdecred/godcr/ui/decredmaterial/progressbar.go:160 +0x391
gioui.org/layout.Inset.Layout({0x11c, 0x0, 0x4fd99e0, 0xc0}, {{{0x212, 0x0}, {0x212, 0xf41e5}}, {0x3f800000, 0x3f800000}, ...}, ...)
	/home/amos/go/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.(*MultiLayerProgressBar).Layout.func1({{{0x212, 0x0}, {0x212, 0xf41ea}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, 0xe65ee2f75, ...}, ...})
	/home/amos/go/src/github.com/planetdecred/godcr/ui/decredmaterial/progressbar.go:182 +0x98
gioui.org/layout.Flex.Layout({0x0, 0x3d, 0x3b, 0xc0}, {{{0x212, 0x0}, {0x212, 0xf41fc}}, {0x3f800000, 0x3f800000}, ...}, ...)
	/home/amos/go/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/decredmaterial.(*MultiLayerProgressBar).Layout(0xc000093aa0, {{{0x212, 0x0}, {0x212, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, ...}, ...}, ...)
	/home/amos/go/src/github.com/planetdecred/godcr/ui/decredmaterial/progressbar.go:179 +0x175
github.com/planetdecred/godcr/ui/page/staking.(*Page).walletBalanceLayout.func1.2({{{0x212, 0x0}, {0x212, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, 0xe65ee2f75, ...}, ...})
	/home/amos/go/src/github.com/planetdecred/godcr/ui/page/staking/balance_layout.go:73 +0x345
gioui.org/layout.Flex.Layout({0xb0, 0x14, 0x4a, 0xc0}, {{{0x212, 0x0}, {0x212, 0xf4218}}, {0x3f800000, 0x3f800000}, ...}, ...)
	/home/amos/go/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/page/staking.(*Page).walletBalanceLayout.func1({{{0x212, 0x0}, {0x212, 0xf4218}}, {0x3f800000, 0x3f800000}, {0x1d81dc0, 0xc0000ff8e8}, {0xc0a09a969f1c6a80, 0xe65ee2f75, ...}, ...})
`

### LINKED ISSUE #978: Recent tickets list does not update
Recent tickets list does not update on new ticket purchases, only updates after leaving and returning to the page.

### LINKED ISSUE #981: Staking page v2 redesign
Upgrade Staking page to the new v2 layout design.
image

Staking info button.
image

https://www.figma.com/file/e0cYaAzkXyXXPY00v23cQY/GoDCR-%26-DCRMobile?node-id=3650%3A146050

### LINKED ISSUE #1005: Staking window password input field does not handle 'return'
Pressing return to enter a password on the staking window will cause a newline instead of submitting the modal form.

Screen Shot 2022-06-24 at 4 47 45 PM

### LINKED ISSUE #1022: Clicking the staking page twice causes the app to crash.
Clicking on the staking page multiple times causes the app to crash with the following log.
```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x4eb3245]

goroutine 1167 [running]:
github.com/planetdecred/godcr/ui/page/staking.(*Page).loadPageData.func2()
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/staking/overview.go:189 +0xa5
created by github.com/planetdecred/godcr/ui/page/staking.(*Page).loadPageData
	/Users/sirmorrison/Desktop/go projects/godcr-gio/ui/page/staking/overview.go:163 +0xa5```

### PR DESCRIPTION
**Pull request by Sirmorrison**
Resolves #981 
Fix #1014
Fix #976 
Fix #1005
Fix #1022 
Fix #978

- [x] Fix wallet selector page display when app loads for the first time.
- [x] Fix backup modal notification display for wallet that don't have their seed phrase backed up.
- [x] Update the staking layout to the v2 layout design
- [x] Update stake shuffle and staking page to single wallet selector approach (needed to test the account selector)
- [x] Fix crash when staking page button is clicked twice
- [x] Fix ticket list refresh bug
- [x] Fix app crash when the settings button is clicked when a watch only wallet is selected.

image

image

### CODE REVIEW

**Comment by dreacot** [comment]
Description needs to be updated, because from what i'm seeing it looks like the work is incomplete, and that might have been intentional as follow up PRs are planned, but i'm not sure, can the description state the overall expectation of this PR(including what's purposely left out) and it we are expecting follow up PRs on this

**Comment by Sirmorrison** [comment]
> Description needs to be updated, because from what i'm seeing it looks like the work is incomplete, and that might have been intentional as follow up PRs are planned, but i'm not sure, can the description state the overall expectation of this PR(including what's purposely left out) and it we are expecting follow up PRs on this

The PR resolves everything mentioned above. The next PR is the tx details page, and that does not really have anything to do with this PR.

**Review by dreacot — DISMISSED** [review]
1. Navigating to the staking page cause a overall lag of the app for a couple of seconds.

Screenshot from 2022-07-26 16-19-05

2. All tickets are showing at once which isn't efficient and most likely the cause of the lag. the list should be paginated.

3. clicking on the settings icon on the staking page causes a crash

4. The statistic dialog is no same as that of the desing
Screenshot from 2022-07-26 16-24-31
Screenshot from 2022-07-26 16-25-28

Screenshot from 2022-07-26 16-26-54
5. there should be more right margin after the settings icon, and `stake` should be in bold

Screenshot from 2022-07-26 16-28-58
6. This section should be titled`Tickets` and the icons don't seem to be on par with the provided figma design

7. the hover on tickets doesn't have a proper padding
Screenshot from 2022-07-26 16-30-25
Screenshot from 2022-07-26 16-30-17

8. The staking details page doesn't yet seem to be on par with the figma design

**Comment by Sirmorrison** [comment]
> 2\. All tickets are showing at once which isn't efficient and most likely the cause of the lag. the list should be paginated.

* `clicking on the settings icon on the staking page causes a crash` attach log please
* `All tickets are showing at once which isn't efficient and most likely the cause of the lag. the list should be paginated.` for this, we will need to open a new PR
probably fix with the tx page. #939
* `The statistic dialog is no same as that of the desing` there is a default godcr modal design which was used. Having to implement this design means changing the godcr modal design which wont be uniform. Was to bring it up on the design group.. Skipped my mind. 
* `The staking details page doesn't yet seem to be on par with the figma design` this is coming in a different PR as the design is a tx details design overhaul.

**Comment by dreacot** [comment]
> clicking on the settings icon on the staking page causes a crash attach log please
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x120a09b]

goroutine 14 [running]:
github.com/planetdecred/godcr/ui/page/components.(*AccountSelector).Layout.func2({{{0x0, 0x0}, {0x114, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/components/account_selector.go:181 +0x3b
gioui.org/layout.Flex.Layout({0xfa, 0x3d, 0x0, 0x0}, {{{0x0, 0x0}, {0x134, 0xf41fc}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1.1.1({{{0x0, 0x0}, {0x134, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:89 +0x85
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x134, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:166 +0x162
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1.1({{{0x134, 0x0}, {0x134, 0xf41fc}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:88 +0xe5
gioui.org/layout.Inset.Layout({0x1bd0120, 0x0, 0x0, 0x80000000}, {{{0x134, 0x0}, {0x134, 0xf41fc}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1({{{0x14c, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:86 +0x109
github.com/planetdecred/godcr/ui/decredmaterial.Border.Layout({{0xe6, 0xea, 0xed, 0xff}, {0x8, 0x8, 0x8, 0x8}, 0x40000000}, {{{0x14c, ...}, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/border.go:21 +0x98
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2({{{0x14c, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:84 +0x1d6
gioui.org/layout.Stack.Layout({0xff}, {{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/stack.go:60 +0x8b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1({{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:47 +0x2aa
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1({{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:108 +0x238
gioui.org/layout.Inset.Layout({0x417c8a, 0x0, 0x41a154, 0x0}, {{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1({{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:44 +0x109
gioui.org/layout.Direction.Layout(0x0, {{{0x0, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:166 +0x162
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout({0xfffffffffffffffe, 0xffffffffffffffff, 0x0, {0x0, 0x0, 0x0, 0x0}, 0x0, {{0xe6, 0xea, ...}, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:42 +0xe5
github.com/planetdecred/godcr/ui/page/components.(*AccountSelector).Layout(0xc003526f00, {0x1dcc100, 0xc00043d3b0}, {{{0x14c, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/components/account_selector.go:170 +0x4a5
github.com/planetdecred/godcr/ui/page/staking.(*ticketBuyerModal).Layout.func2.1.1({{{0x14c, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/staking/stake_modal.go:126 +0x87
gioui.org/layout.Inset.Layout({0x5d0c, 0x0, 0x2899440, 0xc0}, {{{0x14c, 0x0}, {0x14c, 0xf4214}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/page/staking.(*ticketBuyerModal).Layout.func2.1({{{0x14c, 0x0}, {0x14c, 0xf422c}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/staking/stake_modal.go:125 +0xaf
gioui.org/layout.Flex.Layout({0x0, 0x0, 0x0, 0x0}, {{{0x14c, 0x0}, {0x14c, 0xf422c}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/page/staking.(*ticketBuyerModal).Layout.func2({{{0x14c, 0x0}, {0x14c, 0xf422c}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/staking/stake_modal.go:120 +0x255
gioui.org/layout.Inset.Layout({0x3df6, 0x0, 0x3df5, 0x0}, {{{0x14c, 0x0}, {0x14c, 0xf422c}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.2.2({{{0x14c, 0x0}, {0x14c, 0xf4240}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/modal.go:167 +0x85
gioui.org/layout.(*List).Layout(0xc0061bd800, {{{0x14c, 0x0}, {0x14c, 0xf4240}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/list.go:117 +0x17c
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.2({{{0x14c, 0x0}, {0x14c, 0x196}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/modal.go:166 +0x2bd
gioui.org/layout.Flex.Layout({0xf2, 0x3d, 0x0, 0x0}, {{{0x0, 0x0}, {0x14c, 0x1c4}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/flex.go:99 +0xdb5
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1.1.1({{{0x0, 0x0}, {0x14c, 0x1c4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:89 +0x85
gioui.org/layout.Direction.Layout(0x8, {{{0x0, 0x0}, {0x14c, 0x1c4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:166 +0x162
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1.1({{{0x0, 0x0}, {0x14c, 0x1c4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:88 +0xe5
gioui.org/layout.Inset.Layout({0x0, 0x0, 0x0, 0x0}, {{{0x0, 0x0}, {0x14c, 0x1c4}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2.1({{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:86 +0x109
github.com/planetdecred/godcr/ui/decredmaterial.Border.Layout({{0x0, 0x0, 0x0, 0x0}, {0xe, 0xe, 0xe, 0xe}, 0x0}, {{{0x0, ...}, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/border.go:21 +0x98
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1.2({{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:84 +0x1d6
gioui.org/layout.Stack.Layout({0x40}, {{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/stack.go:60 +0x8b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1.1({{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:47 +0x2aa
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1.1({{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:108 +0x238
gioui.org/layout.Inset.Layout({0x3fb47b13, 0xa, 0x2ed12a0, 0x0}, {{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout.func1({{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:44 +0x109
gioui.org/layout.Direction.Layout(0x8, {{{0x0, 0x0}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:166 +0x162
github.com/planetdecred/godcr/ui/decredmaterial.LinearLayout.Layout({0xffffffffffffffff, 0xffffffffffffffff, 0x1, {0xff, 0xff, 0xff, 0xff}, 0x0, {{0x0, 0x0, ...}, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/linearlayout.go:42 +0xe5
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1({{{0x0, 0x1f4}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/modal.go:130 +0x3ac
gioui.org/layout.Inset.Layout({0xf51688, 0x0, 0xfffffffe, 0xffffffff}, {{{0x0, 0x1f4}, {0x17c, 0x1f4}}, {0x3f800000, 0x3f800000}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/layout.go:139 +0x3b6
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2({{{0x0, 0x258}, {0x17c, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/modal.go:118 +0x2f4
gioui.org/layout.Stack.Layout({0x75}, {{{0x0, 0x0}, {0x320, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/stack.go:60 +0x8b6
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout(0xc0072f8fa0, {{{0x0, 0x0}, {0x320, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/decredmaterial/modal.go:91 +0x2ad
github.com/planetdecred/godcr/ui/page/staking.(*ticketBuyerModal).Layout(0xc005225700, {{{0x0, 0x0}, {0x320, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/page/staking/stake_modal.go:158 +0x185
github.com/planetdecred/godcr/ui.(*Window).prepareToDisplayUI.func3({{{0x0, 0x0}, {0x320, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, 0xa3fb47b13, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/window.go:234 +0x58
gioui.org/layout.Stack.Layout({0x40}, {{{0x320, 0x258}, {0x320, 0x258}}, {0x3f800000, 0x3f800000}, {0x1d7bbe0, 0xc00001f8e8}, {0xc0b020fccc6fe14c, ...}, ...}, ...)
        /home/kennedy/Projects/pkg/mod/gioui.org@v0.0.0-20220601100144-a896a467ecae/layout/stack.go:60 +0x8b6
github.com/planetdecred/godcr/ui.(*Window).prepareToDisplayUI(0xc0000f5d60, {{0xc0b020fccc6fe14c, 0xa3fb47b13, 0x2ed12a0}, {0x3f800000, 0x3f800000}, {0x320, 0x258}, {0x0, 0x0, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/window.go:241 +0x42d
github.com/planetdecred/godcr/ui.(*Window).handleFrameEvent(0xc0000f5d60, {{0xc0b020fccc6fe14c, 0xa3fb47b13, 0x2ed12a0}, {0x3f800000, 0x3f800000}, {0x320, 0x258}, {0x0, 0x0, ...}, ...})
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/window.go:182 +0x227
github.com/planetdecred/godcr/ui.(*Window).HandleEvents(0xc0000f5d60)
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/ui/window.go:148 +0x156
main.main.func2()
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/main.go:80 +0x26
created by main.main
        /home/kennedy/Projects/src/github.com/planetdecred/godcr/main.go:79 +0x4d2
exit status 2
```

**Review by dreacot — COMMENTED** [review]
clicking the settings icon isn't working, is that intentional?

**Comment by Sirmorrison** [comment]
> clicking the settings icon isn't working, is that intentional?

if you are using a watch only wallet, yes it wont work as its watch only. but other wallets works.

**Comment by dreacot** [comment]
> > clicking the settings icon isn't working, is that intentional?
> 
> if you are using a watch only wallet, yes it wont work as its watch only. but other wallets works.

alright, there's nothing indicating that, maybe we should use a toast?

**Comment by JustinBeBoy** [comment]
> > > clicking the settings icon isn't working, is that intentional?
> > 
> > 
> > if you are using a watch only wallet, yes it wont work as its watch only. but other wallets works.
> 
> alright, there's nothing indicating that, maybe we should use a toast?

or shouldn't show it

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
1. After confirm automatic Ticket Purchase the switcher not turn on then turn off without notice for user the reason. 

2. Should uppercase the first character
image

File: ui/page/staking/stake_info.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+										if pg.WL.MultiWallet.IsSyncing() {
+											txt.Text = values.String(values.StrSyncingState)
+										}
+										return txt.Layout(gtx)
+									}),
+								)
+							}),
+							pg.dataRows(values.String(values.StrLiveTickets), pg.ticketOverview.Unmined),
+							pg.dataRows("Can Buy", pg.CalculateTotalTicketsCanBuy()),
```
Comment (JustinBeBoy) [inline ui/page/staking/stake_info.go]: Add "Can Buy" to the localize string

**Comment by JustinBeBoy** [comment]
We don't use description of ticket like mockup ?

image

**Comment by Sirmorrison** [comment]
* After confirm automatic Ticket Purchase the switcher not turn on then turn off without notice for user the reason.
this seems to be a backend issue and will be resolved from dcrlibwallet. https://github.com/planetdecred/dcrlibwallet/issues/258


> We don't use description of ticket like mockup ?
> 
> image

I dont understand this comment please.

Can you share a screenshot of the UI that did not align?

**Comment by dreacot** [comment]
LGTM @JustinBeBoy are you still having concerns?

**Review by JustinBeBoy — APPROVED** [review]

**Pull request merged**